### PR TITLE
[DEM-710] Refactor SDK to conform to qiskit 0.7.0

### PR DESCRIPTION
**Pull request by qSaevar**

### CODE REVIEW

File: src/tests/quantuminspire/qiskit/test_backend_qx.py [hunk 1]
```diff
@@ -181,8 +182,8 @@ def test_validate_OperationAfterMeasure(self):
     def test_no_operation_after_measure_cx_gate(self):
         with patch.object(QuantumInspireBackend, "_submit_experiment", return_value=Mock()):
             simulator = QuantumInspireBackend(Mock(), Mock(), logger=Mock())
-            instructions = [{'name': 'X', 'qubits': [1]}, {'name': 'measure', 'qubits': [0], 'clbits': [0]},
-                            {'name': 'CX', 'qubits': [0, 1], 'clbits': [0, 1]}]
+            instructions = [{'name': 'X', 'qubits': [1]}, {'name': 'measure', 'qubits': [0], 'memory': [0]},
+                            {'name': 'CX', 'qubits': [0, 1], 'memory': [0, 1]}]
```
[the remark this answers is not in the document]
Comment (qSaevar) [reply src/tests/quantuminspire/qiskit/test_backend_qx.py]: nothing in this case

File: docs/qi-performance-test.ipynb [hunk 2]
```diff
@@ -82,13 +82,14 @@
...
...9qmk+HcOTgVpHYbi4uKiLl26pMrKSvl8PrW3tyudTm/4tQEAKGRVFUHt37PbuG9rdaX21H3l8ESQ1mEoJhIJPXjwQCMjI3r1aulTTqdPn97wawMAUOi+aW7UlyXF723/+98+fLYR+eNaKD579kzRaFRVVVUKBAI6fvy4JGlgYECxWEy1tbUqKyvT1atXlUwmNTExkfeZ3FwbAIBCt+n3X+qb5sYV2+p37dC2rdUuTQRXQvHu3bs6cuSIuru7NTk5qampKXV1dWl2dlaTk5NqbPzll2THjh3y+/16/vz5R62RSCQUDofXfHwu1wYAAL9N0546Vf6xTNLS7XAORfa5PFFhczwU5+fnde7cOV27dk3t7e0qKSmRz+dTNBqVZVmSpLKyshU/EwgEZFmW0um0mpqaFIlEtHfvXj158mTVdXp6ej4q8LKtDQAA8u/d2+Uc2Bvm5touc/ybWYaHh+XxeNTZ2fnePp9v6ZdhdnblTTdnZmbk9/sVDAb19OlTeb1evXz5UidPnlRra2tO5sq2dr719A3kfQ0AAD4n333/g777/ge3xygIiVi3cbvjoZhOpxUMBuUxvCk1EAgoFAppdHRUDQ0NkqTx8XFZlqVwOCyv17t87NzcnOrrc3c/pWxr59tqLxAAAIXItm1jK8BZjl96bmxsVCqV0sOHD/X27VvNzMzo8ePHy/u7u7vV19enVColy7IUi8UUjUa1bds2SVIqlVJzc7MOHjyoo0eP5nS2bGsDAABnEInrgyvf9Tw0NKTe3l69fv1aPp9PZ86cUTwel7R0L8NYLKbBwUFlMhm1trZqYGBAFRUVK55jYmJCLS0tq34iube3V7dv39bY2Nia51rr2vnApWcAAOCW1a5suhKKv1Umk1FJSYkk6c2bN4pEInrx4oXLUwEAAGxMjr9H8VOMjo4qFovJ6/VqYWFB/f39bo8EAACwYX1WZxQBAADgnHX3FX4AAABYHwhFAAAAGBGKAAAAMCIUAQAAYEQoAgAAwIhQBAAAgBGhCAAAACNCEQAAAEaEIgAAAIwIRQAAABgRigAAADAiFAEAAGBEKAIAAMCIUAQAAIARoQgAAAAjQhEAAABGhCIAAACMCEUAAAAYEYoAAAAwIhQBAABgRCgCAADAiFAEAACAEaEIAAAAI0IRAAAARj8Bxt7o/ChMU3AAAAAASUVORK5CYII=\n",
```
[the remark this answers is not in the document]
Comment (qSaevar) [reply docs/qi-performance-test.ipynb]: ✔️

File: src/quantuminspire/qiskit/backend_qx.py [hunk 3]
```diff
@@ -258,12 +269,12 @@ def _collect_measurements(experiment):
             List: A list of lists, for each measurement List contains a list of [qubit_index, classical_bit_index]
         """
         header = experiment.header
-        number_of_qubits = header.number_of_qubits
```
[the remark this answers is not in the document]
Comment (qSaevar) [reply src/quantuminspire/qiskit/backend_qx.py]: ✔️

File: docs/qi-performance-test.ipynb [hunk 4]
```diff
@@ -197,7 +197,7 @@
      "output_type": "stream",
      "text": [
       "Generated histogram:\n",
-      "{'1101': 62.0, '1011': 71.0, '0111': 70.0, '1100': 75.0, '0101': 53.0, '0001': 65.0, '1001': 56.0, '1010': 74.0, '1111': 63.0, '0110': 71.0, '0000': 64.0, '0100': 63.0, '1000': 70.0, '0011': 57.0, '1110': 54.0, '0010': 56.0}\n"
+      "{'1': 63, '1101': 62, '1000': 73, '1100': 53, '1010': 62, '1011': 61, '111': 54, '10': 69, '100': 61, '1111': 73, '1110': 71, '101': 59, '110': 54, '0': 72, '1001': 71, '11': 66}\n"
```
[the remark this answers is not in the document]
Comment (qSaevar) [reply docs/qi-performance-test.ipynb]: I'll find out were the padding went. It's a shame about the order, I can fix it, though that is not the qiskit way: https://github.com/Qiskit/qiskit-tutorials/blob/master/qiskit/basics/the_ibmq_provider.ipynb
But indeed order is nicer. I'll have a look.

**Comment by quantumkoen** [comment]
@qSaevar in addition, the `QuantumInspireProvider.backends()` method should not show any backend types that have `is_allowed` set to False

File: docs/qi-performance-test.ipynb (hunk 4 above)
[the remark this answers is not in the document]
Comment (qSaevar) [reply docs/qi-performance-test.ipynb]: ✔️

File: src/quantuminspire/qiskit/backend_qx.py [hunk 5]
```diff
@@ -15,59 +15,70 @@
...
+    DEFAULT_CONFIGURATION = BackendConfiguration(
+        backend_name='qi_simulator',
+        backend_version=quantum_inspire_version,
+        n_qubits=26,
+        basis_gates=['x', 'y', 'z', 'h', 's', 'cx', 'ccx', 'u1', 'u2', 'u3', 'id', 'snapshot'],
+        gates=[GateConfig(name='NotUsed', parameters=['NaN'], qasm_def='NaN')],
+        conditional=False,
+        simulator=True,
+        local=False,
+        memory=False,
```
[the remark this answers is not in the document]
Comment (qSaevar) [reply src/quantuminspire/qiskit/backend_qx.py]: ✔️

File: docs/grover_algorithm_qi.ipynb [hunk 6]
```diff
@@ -847,15 +829,15 @@
...
+   "execution_count": 17,
    "metadata": {},
    "outputs": [
     {
      "name": "stdout",
      "output_type": "stream",
      "text": [
       "Generated histogram:\n",
-      "{'000': 3, '001': 1, '010': 4, '011': 4, '100': 4, '101': 3, '110': 4, '111': 489}\n"
+      "{'010': 4, '011': 5, '001': 5, '111': 482, '000': 7, '110': 7, '100': 2}\n"
```
[the remark this answers is not in the document]
Comment (qSaevar) [reply docs/grover_algorithm_qi.ipynb]: ✔️

File: docs/grover_algorithm_qi.ipynb [hunk 7]
```diff
@@ -40,7 +40,7 @@
     "\n",
     "from qiskit.tools.visualization import circuit_drawer, plot_histogram\n",
     "from qiskit import execute, QuantumCircuit, QuantumRegister, ClassicalRegister\n",
-    "from qiskit import Aer\n",
+    "from qiskit import BasicAer\n",
```
[the remark this answers is not in the document]
Comment (qSaevar) [reply docs/grover_algorithm_qi.ipynb]: 👍

File: README.md [hunk 8]
```diff
@@ -20,7 +20,13 @@ on cQASM can be found in the Quantum Inspire
 
 ## Installation
 
-The source for the SDK can be found at Github. For the default installation execute:
+The Quantum Inspire SDK can be installed from PyPI via pip:
+
+```
+$ pip install quantuminspire
+```
+
```
Comment (quantumkoen) [inline README.md]: I'd add something like:

````
In addition, to use Quantum Inspire through Qiskit or ProjectQ, install either or both of
the qiskit and projectq packages:

```
$ pip install qiskit
$ pip install projectq
```

````

File: README.md [hunk 9]
```diff
@@ -20,7 +20,13 @@ on cQASM can be found in the Quantum Inspire
...
+The source for the SDK can also be found at Github. For the default installation execute:
```
Comment (quantumkoen) [inline README.md]: Probably good to make clear that this is about installing from source by introducing a header before this line, `### Installing from source`. I would also include git clone instructions, something like:

```
$ git clone https://github.com/QuTech-Delft/quantuminspire
$ cd quantuminspire
$ pip install .
```

**Comment by quantumkoen** [comment]
@qSaevar i was able to run the tests with this command:

```
coverage run --source="./src/quantuminspire" -m unittest discover -s src/tests -t . -v
```

The difference wrt the readme is the -t argument

**Pull request merged**